include/revng/Model/Importer/TypeCopier.h

```diff
@@ -42,23 +42,16 @@ class TypeCopier {
     FromModel(FromModel), DestinationModel(DestinationModel) {}
   ~TypeCopier() { revng_assert(Finalized); }
 
-  model::DefinitionReference copyTypeInto(model::DefinitionReference &Type) {
+  model::UpcastableType copyTypeInto(const model::TypeDefinition &Definition) {
     ensureGraph();
 
-    revng_assert(Type.isValid());
-
-    model::DefinitionReference Result;
-    llvm::df_iterator_default_set<Node *> VisitedFromTheType;
-    for (Node *N :
-         depth_first_ext(TypeToNode.at(Type.get()), VisitedFromTheType))
+    model::UpcastableType Result;
+    llvm::df_iterator_default_set<Node *> Visited;
+    for (Node *N : depth_first_ext(TypeToNode.at(&Definition), Visited))
       ;
 
     for (const auto &P : FromModel->TypeDefinitions()) {
-      if (auto *PrD = llvm::dyn_cast<model::PrimitiveDefinition>(P.get())) {
-        // Ensure we have all the necessary primitive types
-        DestinationModel->getPrimitiveType(PrD->PrimitiveKind(), PrD->Size());
-      } else if (AlreadyCopied.count(P.get()->ID()) == 0
-                 and VisitedFromTheType.contains(TypeToNode.at(P.get()))) {
+      if (Visited.contains(TypeToNode.at(P.get()))) {
```

Comment:
You dropped the `AlreadyCopied.count(P.get()->ID()) == 0`, you sure?

Reply:
Wow, I wonder why. I guess I didn't notice the fact that it supports being called multiple times with the same object. My bad, thanks for noticing!

Reply:
Done!